ndlarflow/h5_to_root_ndlarflow.py

```diff
@@ -0,0 +1,254 @@
+import numpy as np
+import awkward as awk
+
+import uproot as ur
+import h5py as h5
+
+from h5flow.data import dereference
+import h5flow
+import os
+import sys
+
+# Refactored version of h5 to ROOT conversion, Bruce Howard - 2025
+# initial scripts are now saved in e.g. _minirun6_2.py and _minirun6_3.py versions e.g. and thanks to Richie Diurba and any others who made these scripts
+
+# NOTE for more on uproot TTree writing see the below or the uproot documentation
+# see https://stackoverflow.com/questions/72187937/writing-trees-number-of-baskets-and-compression-uproot
+
+# Main function with command line settable params
+def printUsage():
+    print('python h5_to_root_ndlarflow_ndmc_more_refactor.py FileList IsData IsFinalHits OutName')
+    print('-- Parameters')
+    print('FileList    [REQUIRED]:                           comma separated set of files to convert - note it will be one output')
+    print('IsData      [OPTIONAL, DEFAULT = 0, is MC]:       1 = Data, otherwise = MC')
+    print('IsFinalHits [OPTIONAL, DEFAULT = 0, prompt hits]: 1 = use "final" hits, otherwise = "prompt"')
+    print('OutName     [OPTIONAL, DEFAULT = input[0]+stuff]: string for an output file name if you want to override. Note that default writes to current directory.')
+    print('')
+
+def main(argv=None):
+    fileNames=[]
+    useData=False
+    useFinalHits=False
+    overrideOutname=1
+    outname=''
+
+    MeV2GeV=0.001
+    trueXOffset=0 # Offsets if geometry changes
+    trueYOffset=0 #42+268
+    trueZOffset=0 #-1300
+
+    if len(sys.argv)==1:
+        print('---------------------------------------------------------------')
+        print('Must at least pass a file location/name to be converted, usage:')
+        print('---------------------------------------------------------------')
+        printUsage()
+        return
+    if len(sys.argv)>1:
+        if str(sys.argv[1])=='help' or str(sys.argv[1])=='h' or str(sys.argv[1])=='-h' or str(sys.argv[1])=='--help':
+            print('---------------------------------------------------------------')
+            print('usage:')
+            print('---------------------------------------------------------------')
+            printUsage()
+            return
+        elif sys.argv[1]!=None:
+            fileList=str(sys.argv[1])
+            fileNames=fileList.split(',')
+        if len(sys.argv)>2 and sys.argv[2]!=None:
+            if int(sys.argv[2])==1:
+                useData=True
+        if len(sys.argv)>3 and sys.argv[3]!=None:
+            if int(sys.argv[3])==1:
+                useFinalHits=True
+        if len(sys.argv)>4 and sys.argv[4]!=None:
+            outname=str(sys.argv[4])
+            overrideOutname=0
+
+    MaxArrayDepth=int(10000)
+    MaxArrayDepthData=int(100000)
+    isWritten=False
+
+    promptKey='prompt'
+    if useFinalHits==True:
+        promptKey='final'
+
+    if overrideOutname==1:
+        outname = fileNames[0].split('/')[-1]+'_hits.root'
```

Comment:
Should we use `_hits_uproot.root` to follow the convention used for the ROOT macro?

Reply:
Sure!